server/src/main/java/io/crate/analyze/relations/select/SelectAnalyzer.java

```diff
@@ -49,13 +53,50 @@ public static SelectAnalysis analyzeSelectItems(List<SelectItem> selectItems,
                                                     Map<RelationName, AnalyzedRelation> sources,
                                                     ExpressionAnalyzer expressionAnalyzer,
                                                     ExpressionAnalysisContext expressionAnalysisContext) {
+        selectItems = adjustSelectItemsFromAliasedTableFunctions(selectItems, sources);
         SelectAnalysis selectAnalysis = new SelectAnalysis(
             selectItems.size(), sources, expressionAnalyzer, expressionAnalysisContext);
         selectItems.forEach(x -> x.accept(INSTANCE, selectAnalysis));
         SelectSymbolValidator.validate(selectAnalysis.outputSymbols());
         return selectAnalysis;
     }
 
+    private static List<SelectItem> adjustSelectItemsFromAliasedTableFunctions(List<SelectItem> selectItems,
+                                                                               Map<RelationName, AnalyzedRelation> sources) {
+        if (sources.values().stream()
+            .filter(x -> x instanceof AliasedAnalyzedRelation)
+            .noneMatch(y -> ((AliasedAnalyzedRelation) y).relation() instanceof TableFunctionRelation))
+            return selectItems;
+
+        var relNames = sources.keySet().stream().map(RelationName::name).collect(Collectors.toList());
```

Comment:
`relNames` currently does not but would like it to hold all aliases to `AliasedTableFunctionRelation` from the source relations. probably could write my own visitor pattern for that.